File: java/118.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> triangle = new ArrayList<>();

        List<Integer> row1 = new ArrayList<>();
        row1.add(1);
        triangle.add(row1);

        for(int i = 1; i < numRows; i++){
            List<Integer> thisRow = new ArrayList<>();

            thisRow.add(1);

            for(int j = 1; j < triangle.get(i - 1).size(); j++){
                int addValue = triangle.get(i - 1).get(j) + triangle.get(i - 1).get(j - 1);
                thisRow.add(addValue);
            }

            thisRow.add(1);
            triangle.add(thisRow);
        }

        return triangle;
    }
}

class Main {
    public static void main(String[] args) {
        Solution solution = new Solution();

        int[] inputs = { 1, 2, 3, 4 };
        for(int rows: inputs){
            System.out.println(solution.generate(rows));
        }

    }
}

File: kotlin/118.kt
class Solution {
    fun generate(numRows: Int): List<List<Int>> {

        // root of the triangle
        val pascalsTriangle = mutableListOf(
            listOf(1)
        )

        // start building additional rows
        for(i in 1 until numRows){

            // start a new row, rows always start with 1
            val nextRowOfTriangle = mutableListOf<Int>(1)

            // loop over the previous row starting at the 2nd index
            for(j in 1 until pascalsTriangle[i-1].size){

                // add the value of the previous index and the current and add to the new row
                nextRowOfTriangle.add(
                    pascalsTriangle[i-1][j-1] + pascalsTriangle[i-1][j]
                )
            }

            //rows always end with 1
            nextRowOfTriangle.add(1)

            // add the new row to the triangle
            pascalsTriangle.add(nextRowOfTriangle)
        }

        return pascalsTriangle
    }
}

fun main(){
    val levels: Int = 10
    val solution = Solution()

    for(i in 1..levels){
        println(solution.generate(i))
    }
}